r/src/array_to_vector.cpp

```diff
@@ -133,6 +146,16 @@ class Converter {
 
  protected:
   std::shared_ptr<ChunkedArray> chunked_array_;
+
+ private:
+  bool CanAltrep(const std::shared_ptr<Array>& array) {
+    if (array->null_count() == 0) {
+      return true;
+    }
+
+    return array->data().use_count() == 1 && array->data()->buffers[1].use_count() == 1 &&
+           array->data()->buffers[1]->is_mutable();
```

Comment:
I thought arrays were immutable? when/how likely is this to be true?

Reply:
specifically here when mutating does not logically change anything. it's my understanding that what's there is undefined and that it is inconsequential to change it. 

it may be false however when the buffer comes from a source that prevents mutation, I don't have a specific example but I guess reading from disk or something.

Reply:
Hmm, yes, I guess it _should_ be logicall inconsequential. OTOH, it may come as a surprise if people e.g. blindlessly memory-map a file in read-write mode without imagining that RArrow will modify it.